default.nix

```
@@ -109,10 +112,11 @@ let
         pkg-def = plan-pkgs.pkgs;
         # The compiler referenced in the stack config
         compiler = (plan-pkgs.extras hackage).compiler or (pkg-def hackage).compiler;
+        patchesModule = ghcHackagePatches.${compiler.nix-name} or {};
```

I wonder if this will eventually bite us. Don't have a much better idea though.

---

Possibly this should emit a trace warning? i.e. if we don't have patches for that compiler it's probably fine, but worth checking that it's deliberately that way. And we could silence the warning in `haskell.nix` by putting in an empty set if that's appropriate.

---

```suggestion
        patchesModule = ghcHackagePatches.${compiler.nix-name} or (builtins.trace "[WARN] no patch module found for ${compiler.nix-name}") {};
```
 something like that?

---

Maybe
`WARNING: no patch module found for ${compiler.nix-name}. This is probably benign, but please report it on the 'haskell.nix' issue tracker.`